File: examples/C#/rrserver.cs
//
//  Request-reply service
//  Connects REP socket to tcp://localhost:5560
//  Expects "Hello" from client, replies with "World"
//

//  Author:     Michael Compton, Tomas Roos
//  Email:      michael.compton@littleedge.co.uk, ptomasroos@gmail.com

using System;
using System.Text;
using ZMQ;

namespace ZMQGuide
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            using (var context = new Context(1))
            {
                using (Socket socket = context.Socket(SocketType.REP))
                {
                    socket.Connect("tcp://localhost:5560");

                    while (true)
                    {
                        string message = socket.Recv(Encoding.Unicode);
                        Console.WriteLine("Received request: " + message);
                        socket.Send("World", Encoding.Unicode);
                    }
                }
            }
        }
    }
}

File: examples/Clojure/rrserver.clj
(ns rrserver
  (:refer-clojure :exclude [send])
  (:require [zhelpers :as mq]))

;;                                                 
;; Hello World server                               
;; Connects REP socket to tcp://*:5560              
;; Expects "Hello" from client, replies with "World"
;; Isaiah Peng <issaria@gmail.com>
;;

(defn -main []
  (let [ctx (mq/context 1)
        responder (mq/socket ctx mq/rep)]
    (mq/connect responder "tcp://localhost:5560")
    (while true
      (let [string (mq/recv-str responder)]
        (println (format "Received request: [%s]." string))
        ;; Do some work
        (Thread/sleep 1000)
        (mq/send responder "World\u0000")))
    ;; We never get here but clean anyhow
    (.close responder)
    (.term ctx)))
